radix-engine/src/system/system_modules/execution_trace/module.rs

```diff
@@ -461,13 +461,16 @@ impl<ModuleApi: SystemModuleApiFor<Self> + SystemModuleApiResourceSnapshotExtens
     SystemModule<ModuleApi> for ExecutionTraceModule
 {
     fn on_create_node(api: &mut ModuleApi, event: &CreateNodeEvent) -> Result<(), RuntimeError> {
+        if api.current_stack_id_uncosted() != 0 {
```

Comment:
MINOR: I'm not a big fan of arbitrary comparisons to `0`.

Perhaps we could have an `is_transaction_intent_thread()` system function?
Or even better, we could have this be `current_stack` which returns some data, e.g. an `is_root` flag, or an `is_contingent` flag or some enum or something?

Reply:
There is no "transaction intent" concept in the system yet. Not sure that will do any good.